roseau/load_flow/models/lines/parameters.py

```diff
@@ -1058,15 +1214,25 @@ def _check_matrix(self) -> None:
                 continue
             # Check that the off-diagonal element have a zero real part
             off_diagonal_elements = matrix[~np.eye(*matrix.shape, dtype=np.bool_)]
-            if not np.allclose(off_diagonal_elements.real, 0):
+            if not self._off_diag_resistance_allowed and not np.allclose(off_diagonal_elements.real, 0):
                 msg = (
                     f"The {matrix_name} matrix of line type {self.id!r} has off-diagonal elements "
                     f"with a non-zero real part."
                 )
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=code)
+                warnings.warn(msg, category=UserWarning, stacklevel=3)
             # Check that the real coefficients are non-negative
             if (matrix.real < 0.0).any():
                 msg = f"The {matrix_name} matrix of line type {self.id!r} has coefficients with negative real part."
                 logger.error(msg)
                 raise RoseauLoadFlowException(msg=msg, code=code)
+
+    @classmethod
+    @contextlib.contextmanager
+    def _allow_off_diag_resistance(cls) -> Iterator[None]:
+        """Lines imported from PwF or OpenDSS usually have non-zero off diagonal resistances."""
+        old_off_diag_resistance_allowed = cls._off_diag_resistance_allowed
+        try:
+            cls._off_diag_resistance_allowed = True
+            yield
+        finally:
+            cls._off_diag_resistance_allowed = old_off_diag_resistance_allowed
```

Comment:
The more I read it, the more I think this is complicated for nothing. Warn the user every time there is resistance on the off-diagonal elements looks sound and safe. Even if it comes from openDSS.